File: Sliding Window/Number of Substrings With Only 1s/solution.cpp
class Solution {
public:
    int numSub(string s) {
        int result = 0, cur = 0, MOD = 1e9 + 7;
        for(char c: s) {
            if(c == '1') {
                ++cur;
                result = (result + cur) % MOD;
            } else cur = 0;
        }
        return result;
    }
};

File: Sliding Window/Number of Substrings With Only 1s/solution.ts
function numSub(s: string): number {
    let result = 0
    let count = 0
    for(let i = 0; i < s.length; i++) {
        if(s[i] === "1") count++
        else if(count > 0) {
            result += ((count) * (count + 1) / 2) % (1e9 + 7)
            result = result % (1e9 + 7)
            count = 0
        }
    }
    result += ((count) * (count + 1) / 2) % (1e9 + 7)
    result = result % (1e9 + 7)
    return result
};